### DISCUSSION TITLE
Q: Exists a Tasmota Official Android App? A: NO!

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16172] [Q&A] [answered]
I don't have an account on Google play and never want one. 
Is there somewhere else I can download the official (if there is one) app safely?

**Comment by barbudor** [comment, marked as answer]
There is no official app

**Reply by sfromis** [reply]
To have an app interface, you should generally aim for an app for your home automation backend of choice, instead of expecting the app to be specifically for Tasmota devices.

**Reply by dac2020** [reply]
I don't have any yet, I'm just playing around with tasmota right now. 
I started looking for a simple 4 channel relay I could control via TCP commands to activate sprinkler valves and that led me to this project which got me curious. Since I just got this new relay working with tasmota, I just wanted to see how I could control it using my phone like videos I've seen.

**Reply by dac2020** [reply]
Oh and really, I was asking if there is an 'official' app, one that doesn't come from some unknown source where you have to signs up to things and give your Google account and all that.

Is it this?
https://www.tasmotrol.com/en/

**Reply by dac2020** [reply]
I got the above installed on my phone and it works perfectly. Just wanted to see how things work. 
Didn't have to sign up to anything which is good.

**Reply by barbudor** [reply]
There is no official app
Someone has made an app and named it tasmotrol
That doesn't make it official

If you are concern about where this app come from, who made it, I suggest to ask is that open source (like tasmota)? Can someone have access to the source code to see if it is safe to use?

There are other PC application in the tasmota ecosystem like Tasmotizer, Tasmota Device Manager, TasmoAdmin...
Although the first 2 have been designed by a member of Tasmota core développer, none of those are "official" but they are open source and any one can check what they are doing

The only official Tasmota app is tasmota itself and its web gui

**Reply by barbudor** [reply]
It's not because an app didn't asked you anything that it doesn't spy on you or open a backdoor

**Reply by dac2020** [reply]
Thank you very much. I'll look into all of these things.

**Reply by dac2020** [reply]
> It's not because an app didn't asked you anything that it doesn't spy on you or open a backdoor

I always thought that apps are required to ask for permission before being allowed access to anything on your phone.

**Reply by barbudor** [reply]
Apps that you install out of GPlay?
Even on GPlay it happens that apps are forcefully removed by Google for 'illicit' behavior
While Android adds more and more controls, many people just click on yes when permissions are requested especially if the app refuse to work without

Even  android itself is spying on you. At some point you have to accept that or live off-grid
That how our life's are now

**Reply by dac2020** [reply]
I know that no matter the app or no app, Android is spying on us :).
The thing is, we can still try to limit those things a little bit by not installing all kinds of garbage on our phones. 
I rarely install anything but definitely want local wifi mobile access to what ever I'll end up doing with tasmota.
I'd like to find an app I can download directly and install it on my old phones, tablets, what ever I need locally. I tend to keep old devices around the house and use them to view security cams and other devices in the house.

**Reply by dac2020** [reply]
So I guess the question really remains. Which Android app is the most trusted by this community and which can be downloaded directly without having to use Google/Apple stores?

**Reply by sfromis** [reply]
None can be called "trusted", and the crappy security of download outside of app stores is not "interesting"....

**Reply by dac2020** [reply]
Fine but that's not what I'm asking. Surely, long time members here are using one more than others so I'm asking which that is.
And, is there one or more that can be downloaded directly instead of through an app site.

**Reply by sfromis** [reply]
I suspect that most long time members are using none of what you ask for. And apps downloadable without the security of app stores are uncommon. To each their own, of course, but do not assume that what you ask for is a common use case.

My mobile access to control Tasmota devices is through a browser, via my backend smarthome setup, or occasionally direct to one device, where a few smartphone shortcuts can make access quick.

**Reply by dac2020** [reply]
If I assumed so many things, I would not be asking questions here of those who know this stuff well.
Maybe I should not use the word 'assume' in these forums since really, I'm just asking and you seem to take that very literally :).

Yes, true, browser even better than an app but I've not seen the browser in action yet so don't know what to expect. 
I will have to look at some previous documentation to get some clues on how to use the browser now that the template is correctly installed.

**Reply by dac2020** [reply]
Got it. Found some info and am able to control it using both browser and curl. 
Just digging for more info on commands but it's all good now and no need for an app as you said :)

**Reply by barbudor** [reply]
There is no real need for a Tasmota app because tasmota is not meant to be controlled individually at device level, except for configuration 

Tasmota is meant to be integrated through MQTT into an home automation system which could provide eventually an app in addition to its built-in web interface

People would use the like of Home Assistant, OpenHAB, Iobroker, Domoticz...
Some hard-core users build their system from elementary pieces such mosquitto as mqtt broker and node-red for automation with eventually a database such as Influxdb and visualization tool such as Grafana

A Tasmota dedicated tool such as Tasmota Device Manager is useful for device maintenance and is the only need for something tasmota specific.

**Reply by dac2020** [reply]
Ha, that's great. I had not once come across this device manager. I love it. Thanks for another nice lead. So much to learn :).

I don't have any home automation system. I used to use something called X10 but now have a hundred of those devices in a box collecting dust. Not yet sure what I'd like to do as I mainly was looking for something I could use here and there that was wireless. Will figure that part out as I play with it. Need to order more tasmota enabled devices I think.

**Reply by brindisi** [reply]
> There is no real need for a Tasmota app because tasmota is not meant...

please excuse, but THAT has to be left up to the user who knows what HE needs.

e.g. ME! 🙂
I would have given an answer like: sure one can find some apps for conrolling tasmota "directly" (@ google playstore). But regarding to the concept of tasmota one might be better off with one off with one of the MQTT apps you also can find there, cause this was the more direct way.

regards

**Reply by sfromis** [reply]
What *you'd like to have* does not change the answer.

**Comment by wealthwallet** [comment]
I ran into this thread while building exactly that, so: I've made TasmoShelf,
an iOS + Android app for Tasmota devices. To be upfront, it's a paid app
(free for up to 3 devices, 9.99 € once for more) and I'm the developer.

What it does differently from the existing apps:

- Discovery that actually finds things. Tasmota doesn't announce itself via
  mDNS in official builds (USE_DISCOVERY is compiled out), so the app sweeps
  your subnet twice. Devices in Wi-Fi power saving answer a ping instantly
  but take seconds to accept a connection, and a single fast pass misses
  exactly those. You can also set a custom IP range for IoT VLANs, or add a
  device by IP.
- Controls come from what the device reports: relays with your own relay
  names, colour/CT/dimmer for lights, shutters with live position, fan
  steps, energy monitoring with an hour of history, and every sensor the
  device publishes — including ones the app has never seen.
- Apple Home via Matter. Tasmota 13+ on an ESP32 already speaks Matter on
  its own; the app reads chip and firmware, tells you which of your devices
  can join Apple Home directly, opens pairing and shows the setup code. An
  ESP8266 can't do Matter itself, but a single ESP32 running Tasmota
  bridges the old ones in, and the app tells you which devices that
  concerns.
- Nothing leaves your phone. No account, no cloud, no telemetry, and the
  licence is checked offline, so you never have to be online to keep it
  working.

It talks plain HTTP to each device, so nothing needs to be installed or
changed on the Tasmota side. Android is a direct APK download from the
site; iOS is currently in App Store review. Details: https://tasmoshelf.app

**Reply by dac2020** [reply]
Years back. I had found an app called  tasmotrol that sounds like yours that I still use. It was free at the time, probably still is. I wanted a way to manually turn tasmota connected devices on and off, not just automated.

It scans for local devices or you add them manually. Then it shows on/off buttons depending on the device.

Question for you. How hard was it to submit your app/s to apple/android? Is it a complex process or pretty simple and where do you do that if you can share some of the urls?

**Reply by dacCR** [reply]
Not official of course but it works.

**Reply by kmal808** [reply]
Sounds like a nice tool. Do you have the app in Testflight and if yes are you accepting testers? Would be interested in contributing to the project also, whether that be code, documentation, infra/ops, testing, etc. Look forward to checking out TasmoShelf.
